setup.py

```diff
@@ -69,6 +70,18 @@
         raise RuntimeError('ERROR: Cython not found.  Exiting.\n       '
                            'Install Cython or don\'t use "--with-cython"')
 
+if 'pip-' in sys.argv[3] and (sys.argv[1] in ['egg_info', 'install']):
```

Comment:
Instead of this trying to see whether pip is used or not, I would remove the `if '--without-cython'` / `if '--with-cython'` checks. 
(and for now maybe set `use_cython` equal to `has_cython`, which can later be fixed to enable running pip on distributed sources that do not need cython)